File: Strike/C++/Trade.cpp
#include "Trade.h"
#include "Order.h"

Trade::Trade(const uint32_t amount, Decimal price, Order buyOrder, Order sellOrder) 
{
  Trade::amount = amount;
  Trade::price = price;
  Trade::stockId = buyOrder.getStockId();
  Trade::buyerId = buyOrder.getBrokerId();
  Trade::sellerId = sellOrder.getBrokerId();
  Trade::buyOrderId = buyOrder.getId();
  Trade::sellOrderId = sellOrder.getId();
}

Trade::Trade() { }

uint32_t Trade::getAmount() const
{ 
  return amount; 
}
  
void Trade::setAmount(const uint32_t amount) 
{ 
  Trade::amount = amount; 
}

Decimal Trade::getPrice() const
{ 
  return price; 
}
  
void Trade::setPrice(Decimal price) 
{ 
  Trade::price = price; 
}

uint32_t Trade::getStockId() const
{ 
  return stockId; 
}
  
void Trade::setStockId(const uint32_t stockId) 
{ 
  Trade::stockId = stockId; 
}

uint32_t Trade::getBuyerId() const
{ 
  return buyerId; 
}
  
void Trade::setBuyerId(const uint32_t buyerId) 
{ 
  Trade::buyerId = buyerId; 
}

uint32_t Trade::getSellerId() const
{ 
  return sellerId; 
}
  
void Trade::setSellerId(const uint32_t sellerId) 
{ 
  Trade::sellerId = sellerId; 
}

uint64_t Trade::getBuyOrderId() const
{ 
  return buyOrderId; 
}
  
void Trade::setBuyOrderId(const uint64_t buyOrderId) 
{ 
  Trade::buyOrderId = buyOrderId; 
}

uint64_t Trade::getSellOrderId() const
{ 
  return sellOrderId; 
}
  
void Trade::setSellOrderId(const uint64_t sellOrderId) 
{ 
  Trade::sellOrderId = sellOrderId; 
}

File: Strike/CS/StockMarketSimulator/DataTypes/Trade.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace StockMarketSimulator
{
  public class Trade
  {
    private int id;

    private int amount;

    private decimal price; 

    private int stockId; 

    private int buyerId;

    private int sellerId;

    private int buyOrderId;

    private int sellOrderId;

    private DateTime executed;

    public Trade() { }

    public Trade(int amount, decimal price, Order buyOrder, Order sellOrder)
    {
      this.amount = amount;
      this.price = price;
      this.stockId = buyOrder.StockId;
      this.buyerId = buyOrder.BrokerId;
      this.buyOrderId = buyOrder.Id;
      this.sellerId = sellOrder.BrokerId;
      this.sellOrderId = sellOrder.Id;
      this.executed = DateTime.Now;
    }

    public int Id
    {
      get { return id; }
      set { id = value; }
    }

    public int Amount
    {
      get { return amount; }
      set { amount = value; }
    }

    public decimal Price
    {
      get { return price; }
      set { price = value; }
    }

    public int StockId
    {
      get { return stockId; }
      set { stockId = value; }
    }

    public int BuyerId
    {
      get { return buyerId; }
      set { buyerId = value; }
    }

    public int SellerId
    {
      get { return sellerId; }
      set { sellerId = value; }
    }

    public int BuyOrderId
    {
      get { return buyOrderId; }
      set { buyOrderId = value; }
    }

    public int SellOrderId
    {
      get { return sellOrderId; }
      set { sellOrderId = value; }
    }

    public DateTime Executed
    {
      get { return executed; }
      set { executed = value; }
    }
  }
}
